src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureBlobAction.java

```diff
@@ -1,34 +1,27 @@
 package com.microsoftopentechnologies.windowsazurestorage;
 
+import com.microsoftopentechnologies.windowsazurestorage.WAStoragePublisher.WAStorageDescriptor;
+import com.microsoftopentechnologies.windowsazurestorage.beans.StorageAccountInfo;
+import hudson.model.Run;
+import hudson.model.RunAction;
 import java.io.IOException;
 import java.util.List;
-
 import javax.servlet.ServletException;
-
 import jenkins.model.Jenkins;
-
 import org.acegisecurity.Authentication;
 import org.kohsuke.stapler.StaplerRequest;
 import org.kohsuke.stapler.StaplerResponse;
 
-import hudson.model.AbstractBuild;
-import hudson.model.RunAction;
-import hudson.model.Run;
-
-import com.microsoftopentechnologies.windowsazurestorage.WAStoragePublisher;
-import com.microsoftopentechnologies.windowsazurestorage.WAStoragePublisher.WAStorageDescriptor;
-import com.microsoftopentechnologies.windowsazurestorage.beans.StorageAccountInfo;
-
 
 public class AzureBlobAction implements RunAction {
-	private final AbstractBuild build;
+	private final Run build;
 	private final String storageAccountName;
 	private final String containerName;
 	private final boolean allowAnonymousAccess;
 	private final AzureBlob zipArchiveBlob;
 	private final List<AzureBlob> individualBlobs;
```

Comment:
This might be out of scope, but this should probably be kept in a separate file, compressed in some form, and loaded on demand.  The size of this can actually get reallllly large if the number of artifacts archived gets really big.